satellite/satellitedb/nodeselection.go

```diff
@@ -253,6 +270,10 @@ func (partial partialQuery) asQuery() query {
 	var q strings.Builder
 	var args []interface{}
 
+	var asOf string
+	if partial.asOfSystemTimeEnabled && partial.asOfSystemTimeInterval != 0 {
```

Comment:
You can avoid the duplicate check if you use a single type and then check for the interval.